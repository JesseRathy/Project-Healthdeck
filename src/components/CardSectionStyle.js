import React from 'react';
import { View } from 'react-native';

//This component is used to stylize the different sections of the card no matter
//what is passed to it

const CardSectionStyle = (props) => {
  return (
  <View style={styles.containerStyle}>
    {props.children}
  </View>
  );
};

const styles = {
  containerStyle: {
      borderBottomWidth: 1,
      padding: 5,
      backgroundColor: '#fff',
      justifyContent: 'flex-start',
      flexDirection: 'row',
      borderColor: '#ddd',
      position: 'relative'
  }
};

export {CardSectionStyle};
