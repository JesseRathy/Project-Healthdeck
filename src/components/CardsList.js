import React, { Component } from 'react';
import { ScrollView } from 'react-native';
import axios from 'axios';

import CardDetail from './CardDetail';

//class based component
class CardsList extends Component {
  // Initial state
  state = { cards : [] };
  //Instantly called when this component is called
  componentWillMount() {
    {/* Get request for hhtps we can maybe use this to load all the cards?*/}
    axios.get('https://rallycoding.herokuapp.com/api/music_albums')
    .then(response => this.setState({ cards: response.data }));
  }
  //{card.title} is how you get a variable from jsx between the text tags
  renderCards() {
    {/* The key in the text tag is how we keep track of what we are rendering! this can be useful
      for when we have a database ready to choose what cards to render on the screen*/}
  return this.state.cards.map(card =>
    <CardDetail key={card.title} card={card} />
    );
  }
  render() {
    console.log(this.state);
    return (
      <ScrollView>
        {this.renderCards()}
      </ScrollView>
    );
  }
}

export {CardsList};
