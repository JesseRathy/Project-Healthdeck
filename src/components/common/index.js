export * from './Button';
export * from './CardStyle';
export * from './CardSectionStyle';
export * from './Header';
export * from './Input';
export * from './Spinner';
