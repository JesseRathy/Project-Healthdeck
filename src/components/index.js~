export * from './Button';
export * from './CardDetail';
export * from './CardsList';
export * from './CardSectionStyle';
export * from './CardStyle';
export