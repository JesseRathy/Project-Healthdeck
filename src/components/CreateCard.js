import React, { Component } from 'react' ;
import { Picker, Text } from 'react-native';
import { connect } from 'react-redux' ;
import { cardUpdate, cardCreate } from '../actions';
import { CardStyle, CardSectionStyle, Input, Button } from './common'



//This functionality should be called when the QR scanner is scanning but
// at the moment I have connected that functionality to an add button to simulate
// the ability to "addACard" to a users deck
class CreateCard extends Component {

//This is the functionality that was added to the "Add" button which ads a card
//to the users deck but, this logic will be called onQRScanned rather than a save button in the future
    onButtonPress () {
        const { name, bio, pro, image, link } = this.props;

        this.props.cardCreate({ name, bio, pro: pro || 'Doctor', image, link });
    }

  render () {
    console.log(this.props.myCards);
    return (
      <CardStyle>
        <CardSectionStyle>
        <Input
          label="Name"
          placeholder="Jane"
          value = {this.props.name}
          onChangeText={ value => this.props.cardUpdate({prop: 'name', value})}
        />
        </CardSectionStyle>

        <CardSectionStyle>
        <Input
          label="Bio"
          placeholder="I like cats."
          value= { this.props.bio}
          onChangeText={ value => this.props.cardUpdate({prop: 'bio', value})}
        />
        </CardSectionStyle>

        <CardSectionStyle>
          <Input
            label="Link"
            placeholder="web.com"
            value= { this.props.link}
            onChangeText={ value => this.props.cardUpdate({prop: 'link', value})}
          />
        </CardSectionStyle>

        <CardSectionStyle>
          <Text style= {styles.pickerTextStyle}> Select Profession</Text>
          <Picker
            style= {{ flex: 1 }}
            selectedValue={this.props.shift}
            onValueChange={value => this.props.cardUpdate({ prop: 'pro' , value })}
          >
            <Picker.Item label="Doctor" value="Doctor" />
            <Picker.Item label="Other" value="Other" />
          </Picker>
        </CardSectionStyle>



        <CardSectionStyle>
          <Button onPress= {this.onButtonPress.bind(this)}>
            Save
          </Button>
        </CardSectionStyle>

      </CardStyle>
    );
  }
}
const styles = {
  pickerTextStyle: {
    fontSize: 18,
    paddingLeft: 20
  }
};
const mapStateToProps = (state) => {
  const { name, bio, pro, image, link } = state.creationForm;

  return { name, bio, pro, image, link } ;
}
export default connect(mapStateToProps, {
  cardUpdate , cardCreate
}) (CreateCard);
