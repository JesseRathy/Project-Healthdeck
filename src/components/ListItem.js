import React, { Component } from 'react';
import { Text, TouchableWithoutFeedback, View, Image } from 'react-native' ;
import { Actions } from 'react-native-router-flux'
import { CardSectionStyle } from './common';

//Retrieves a list of items for a specefic user which is specefied
// through firebase and userId
class ListItem extends Component {

//If the row of the card is pressed then we pass over the props of that card
// to the showCard file which then is rendered due to our Router.js component navigating
// to that screen through the Action method
  onRowPress() {
    Actions.showCard({ myCards: this.props.myCards });
  }

//Render every card
  render () {
    const { name, bio, pro, image } = this.props.myCards;

    return (
      <TouchableWithoutFeedback onPress={this.onRowPress.bind(this)}>
        <View>
          <CardSectionStyle>
            <View style={styles.thumbnailContainerStyle}>
              <Image
               style ={styles.thumbnailStyle}
               source={{uri: image}}
                />
            </View>
            <Text style={styles.titleStyle}>
                {pro} {": "} {name}
            </Text>
          </CardSectionStyle>
        </View>
      </TouchableWithoutFeedback>
    );
  }

}

const styles = {
  titleStyle: {
    fontSize:18,
    paddingLeft:15
  },
  thumbnailContainerStyle: {
    justifyContent: 'center',
    alignItems: 'center',
    marginLeft: 10,
    marginRight: 10
  },
  thumbnailStyle: {
    height: 50,
    width: 50
  }
}
export default ListItem;
