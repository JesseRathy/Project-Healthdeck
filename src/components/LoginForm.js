import React, {Component} from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';
import { emailChanged, passwordChanged, loginUser } from '../actions'
import { CardStyle, CardSectionStyle, Input,Button, Spinner } from './common';

class LoginForm extends Component {

  onEmailChange (text) {
    this.props.emailChanged(text);
  }

  onPasswordChange (text) {
    this.props.passwordChanged(text);
  }

  onButtonPress () {
    const { email , password } = this.props;

    this.props.loginUser ( {email, password }) ;
  }

  renderError() {
    //At default an error is false since its empty
    //if anything is added it will turn true and so it will render
    if (this.props.error) {
      return (
        <View style={{backgroundColor: 'white' }}>
          <Text style={styles.errorTextStyle}>
            {this.props.error}
          </Text>
        </View>
      );
    }
  }

  renderButton() {
    if (this.props.loading) {
      return <Spinner size="large" />;
    }
    return (
      <Button onPress={this.onButtonPress.bind(this)}>
        Login
      </Button>
    );
  }

  render (){
    return (
      <CardStyle>

      <CardSectionStyle>
      <Input
        label="Email"
        placeholder="user@gmail.com"
        onChangeText={this.onEmailChange.bind(this)}
        value={this.props.email}
      />
      </CardSectionStyle>

      <CardSectionStyle>
      <Input
        secureTextEntry
        label="Password"
        placeholder="password"
        onChangeText={this.onPasswordChange.bind(this)}
        value={this.props.password}
      />
      </CardSectionStyle>

      {this.renderError()}

      <CardSectionStyle>
        {this.renderButton()}
      </CardSectionStyle>

      </CardStyle>
    );
  }
}

const styles = {
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red'
  }
}

//A function in which accesses all our reducers which modify our state
// and then we use the redux connect function to update that state
const mapStateToProps = ({auth}) => {
  const { email, password, error, loading } = auth;

  return { email, password, error, loading };
};

export default connect(mapStateToProps, {
  emailChanged, passwordChanged,loginUser
}) (LoginForm);
