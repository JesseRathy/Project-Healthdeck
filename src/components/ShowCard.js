import React, { Component } from 'react' ;
import { Picker, Text } from 'react-native';
import { connect } from 'react-redux' ;
import { cardUpdate, cardCreate } from '../actions';
import { CardStyle, CardSectionStyle, Input, Button } from './common'

class ShowCard extends Component {

    onButtonPress () {
        const { name, bio, pro, image, link } = this.props;
    }

  render () {
  console.log(this.props.myCards);
  const { name, bio, pro, image, link } = this.props.myCards;

    return (
      <CardStyle>
        <CardSectionStyle>
          <Text style={styles.titleStyle}>
              {pro} {": "} {name}
          </Text>
        </CardSectionStyle>

        <CardSectionStyle>
          <Text style={styles.titleStyle}>
              {bio}
          </Text>
        </CardSectionStyle>

        <CardSectionStyle>
          <Text style={styles.titleStyle}>
              {link}
          </Text>
        </CardSectionStyle>

        <CardSectionStyle>
        <Text style={styles.titleStyle}>
            Image should be displayed here having trouble importing it
        </Text>
        </CardSectionStyle>

      </CardStyle>
    );
  }
}
const styles = {
  titleStyle: {
    fontSize:18,
    paddingLeft:15
  }
};
const mapStateToProps = (state) => {
  const { name, bio, pro, image, link } = state.myCards;

  return { name, bio, pro, image, link } ;
}
export default ShowCard;
