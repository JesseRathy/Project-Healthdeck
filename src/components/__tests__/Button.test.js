import {Button} from '../components/common';
import { shallow,configure } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
//Imports to get Enzyme to work if possible -> will work on later


test('renders correctly', () => {
  const wrapper = shallow(<Button text="Test"/>);
  expect(wrapper).toMatchSnapshot();
});
