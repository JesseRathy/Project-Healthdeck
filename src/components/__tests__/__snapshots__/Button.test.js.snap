// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="Test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <Component
        style={
          Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          }
        }
      />,
      "onPress": undefined,
      "style": Object {
        "alignSelf": "stretch",
        "backgroundColor": "#fff",
        "borderColor": "#007aff",
        "borderRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "marginLeft": 5,
        "marginRight": 5,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "style": Object {
          "alignSelf": "center",
          "color": "#007aff",
          "fontSize": 16,
          "fontWeight": "600",
          "paddingBottom": 10,
          "paddingTop": 10,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Component
          style={
            Object {
              "alignSelf": "center",
              "color": "#007aff",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingBottom": 10,
              "paddingTop": 10,
            }
          }
        />,
        "onPress": undefined,
        "style": Object {
          "alignSelf": "stretch",
          "backgroundColor": "#fff",
          "borderColor": "#007aff",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "marginLeft": 5,
          "marginRight": 5,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "style": Object {
            "alignSelf": "center",
            "color": "#007aff",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingBottom": 10,
            "paddingTop": 10,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
